Lib/test/test_asyncio/test_unix_events.py

```diff
@@ -642,12 +643,19 @@ def test_sock_sendfile_exception(self):
         self.assertEqual(1000, self.file.tell())
 
 
-class UnixReadPipeTransportTests(test_utils.TestCase):
+@parameterize
```

Comment:
This is too much magic for a simple need. I suggest to factor out common tests and empty helpers into a base class and then use that to create two test cases, e.g.:

```python

class BaseTests:

    def make_test_protocol(self):
        raise NotImplementedError

    def test_..._1(self):
         ...
         ... self.make_test_protocol() ...


class ProtocolTests(BaseTests, unittest.TestCase):

    def make_test_protocol(self):
        return asyncio.Protocol


class BufferedProtocolTests(BaseTests, unittest.TestCase):

    def make_test_protocol(self):
        return asyncio.BufferedProtocol
```

Reply:
Ok I simplified it a whole lot now :)